File: Java/프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(n, computers, i, visited);
                answer++;
            }
        }

        return answer;
    }

    void dfs(int n, int[][] computers, int depth, boolean[] visited) {
        visited[depth] = true;
        for (int i = 0; i < n; i++) {
            if (computers[depth][i] == 1 && !visited[i]) {
                dfs(n, computers, i, visited);
            }
        }
    }
}

File: Python/프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n

    def dfs(depth):
        visited[depth] = True
        for i in range(n):
            if computers[depth][i] == 1 and not visited[i]:
                dfs(i)

    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1

    return answer